src/box/alter.cc

```diff
@@ -2370,9 +2391,10 @@ on_replace_dd_space(struct trigger * /* trigger */, void *event)
 						     def->fields,
 						     def->field_count);
 		try {
-			(void) new CheckSpaceFormat(alter);
-			(void) new ModifySpace(alter, def);
-			(void) new RebuildCkConstraints(alter);
+			(void)new CheckSpaceFormat(alter);
+			(void)new ModifySpace(alter, def);
+			(void)new RebuildCkConstraints(alter);
+			(void)new RebuildTupleConstraints(alter);
```

Comment:
Can we instead do this?
1. Use the new dictionary in `ModifySpace::alter_def` to create a new space.
2. Replace the space format dictionary with a new one **and** swap field names in `ModifySpace::alter`.

Reply:
Yes, done.